src/rt-app.c

```diff
@@ -469,10 +470,24 @@ shutdown(int sig)
 		}
 	}
 
+	/*
+	 * If we try to interrupt pthread_join() at the end of main then we
+	 * might end up joining the same thread twice which is undefined
+	 * behaviour and can lead to segfaults.
+	 *
+	 * Only join if nowhere else started the join process already.
+	 */
+	if (joining)
+		return;
```

Comment:
Mmm, maybe I'm just being overly cautious, but can't this be racy w.r.t. a sigint happening while main is just after this point? shutdown from signal could see joining still 0, pass this check and join threads again?

Reply:
Yes but FWIW that's adequate enough IMHO. But I just realized though that I set continue_running *after* wakeup_all_threads() which is wrong. So I'll change this to a mutex with trylock since I have to touch the code again anyway :)